high_card.rb

```diff
@@ -1,5 +1,62 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
+ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
 players = []
+cards = []
+
+suits.each do |suit|
+  ranks.each do |rank|
+    cards.push([rank, suit])
+  end
+end
+
+deck = cards.shuffle
+num_of_players = 0
+while (num_of_players < 22)
+  puts 'Enter a player name. If there are not more players, enter "done".'
+  input = gets.chomp
+  if input == "done"
+    break
+  end
+  players.push(input)
+num_of_players += 1
+end
+
+puts "OK! Here's who's in the game: #{players}"
+
+num_of_tries = 0
+while (num_of_tries)
+  num_of_tries += 1
+  puts "When you are ready to play, enter 'play'."
+  ready = gets.chomp
+  if ready == "play"
+    break
+  end
+  if num_of_tries > 3
+    puts "If you want to quit the game, type 'exit'."
+    if ready == 'exit'
+      exit
+    end
+  end
+  redo
```

Comment:
I'd avoid `redo` if you can help it. It makes loops tricky to read, since the reader thinks they know exactly how the loop works, but then there's this one word at the end that changes its meaning.